core/templates/components/summary-tile/learner-story-summary-tile.component.ts

```diff
@@ -78,12 +101,23 @@ export class LearnerStorySummaryTileComponent implements OnInit {
     this.storyLink = this.getStoryLink();
     this.storyTitle = this.storySummary.getTitle();
     this.thumbnailBgColor = this.storySummary.getThumbnailBgColor();
+    let nextIncompleteNode = this.storySummary.getNodeTitles()[
+      this.completedNodeCount];
```

Comment:
Wouldn't this throw an error if all chapters are completed?

Reply:
If all the chapters are completed, the story will be considered completed and in turn, the topic will also be considered complete. We display those stories here which the user selects from the Edit Goals section. Once the topic is completed, we replace the checkbox for the topic in edit goals with a book icon to not set that topic as a Goal again. So, this error won't occur.